service/orchestrator/toe.go

```diff
@@ -382,31 +380,20 @@ func getControls(controls []*orchestrator.Control, levels []string, level string
 		return c, err
 	}
 
-	// Add controls based on their assurance level to the lists low, medium and high. If a controls is not defined regarding the assurance level it is dropped.
-	for i := range controls {
-		switch controls[i].GetAssuranceLevel() {
-		case levels[0]:
-			low = append(low, controls[i])
-		case levels[1]:
-			medium = append(medium, controls[i])
-		case levels[2]:
-			high = append(high, controls[i])
-		default:
+	// Add controls based on their assurance level to the map. If a controls is not defined regarding the assurance level it is dropped.
+	for i, control := range controls {
+		if control.AssuranceLevel == nil {
 			continue
 		}
+		levelControls[controls[i].GetAssuranceLevel()] = append(levelControls[controls[i].GetAssuranceLevel()], control)
 	}
 
 	// Add all needed controls based on the assurance level
-	switch level {
-	case levels[0]:
-		c = append(c, low...)
-	case levels[1]:
-		c = append(c, low...)
-		c = append(c, medium...)
-	case levels[2]:
-		c = append(c, low...)
-		c = append(c, medium...)
-		c = append(c, high...)
+	for i := range levels {
+		c = append(c, levelControls[levels[i]]...)
```

Comment:
So we are intentionally NOT adding the lower levels anymore?

Reply:
The lower levels are added, but the  prerequisite is that the assurance levels are in ascending order. 
I've added the following as comment: 
`// Note: The assurance levels must be sorted in ascending order, e.g., low, medium, high, because the controls of the lower assurance levels must be present in the higger assurance levels. If this is not the case, the controls with assurance levels low will not be included in medium.`